app/models/service_template.rb

```diff
@@ -340,8 +353,76 @@ def provision_action
     resource_actions.find_by(:action => "Provision")
   end
 
+  def update_resource_actions(ae_endpoints)
+    resource_action_options.each do |action|
+      # If the action exists in updated parameters
+      if ae_endpoints[action[:param_key]]
```

Comment:
You only need to find the existing resource_action once before the if/else clause.